### PR TITLE
Update ADAM to version 0.23.0.

### PR DESCRIPTION
**Pull request by heuermh**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by heuermh** [comment]
Ping for review; please let me know if you'd prefer I squash the two commits.

File: recipes/adam/meta.yaml [hunk 1]
```diff
@@ -15,16 +15,16 @@ build:
 
 source:
   fn: adam-distribution-spark2_2.11-{{ version }}-bin.tar.gz
-  url: https://search.maven.org/remotecontent?filepath=org/bdgenomics/adam/adam-distribution-spark2_2.11/0.22.0/adam-distribution-spark2_2.11-0.22.0-bin.tar.gz
-  sha256: 31624954eb473f2ec24354af7e6303a47887cf2ca4076bb71f54618aef59e15b
+  url: https://search.maven.org/remotecontent?filepath=org/bdgenomics/adam/adam-distribution-spark2_2.11/0.23.0/adam-distribution-spark2_2.11-0.23.0-bin.tar.gz
```
Comment (cbrueffer) [inline recipes/adam/meta.yaml]: Since we have {{ version }}, using it in the URL would be good. Makes the next update diff smaller.

**Comment by cbrueffer** [comment]
Merged, thanks Michael!

**Comment by heuermh** [comment]
Thank you, @cbrueffer!

**Pull request merged**